### ISSUE TITLE
Would like to be able to set timeout for models or individual devices

### ISSUE THREAD

**Issue by candlerb** [ytti/oxidized#919]
I am connecting to an ASA5520 and for some reason the ssh connection (at least in debug mode) is very slow: it scrolls past at about 1 screen per second.  It takes more than the default 20 second timeout to get the full config.

I would like to override this at the model level, but have not been able to.  If I use

~~~
timeout: 20
model_map:
  asa5520: asa
models:
  asa5520:
    timeout: 300
~~~

then it still aborts after 20 seconds (and then connects again and retries).

I also tried mapping it to a csv column, with no success.  The only way I can do it at the moment is to increase the global timeout setting.

**Comment by ytti** [comment]
Currently timeout is global only. We'd need to add something like ```resolve_timeout``` here: https://github.com/ytti/oxidized/blob/master/lib/oxidized/node.rb

And then replace all instances of Oxidized.timeout with node.timeout.

**Comment by candlerb** [comment]
Thank you.

FYI, the problem goes away when input debug mode is turned off - the whole connection and config download takes under 10 seconds.

However, with input debug turned on, the execution of "show running-config" takes substantially longer than 20 seconds.

So another solution would be to try to find out why input debug is so slow; after all, it is just appending to a file.  (Is it doing this one character at a time perhaps?)

**Comment by wk** [comment]
The extra delays introduced by having debug on are most likely solved in #1299. 

Closing this issue. further discussion about granular timeout control can continue in #1044.
